From 738ddd3e4c516ce12dd671c149ce642c6be75ce8 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Tue, 11 Nov 2014 10:13:58 +0800
Subject: [PATCH 0522/1074] MLK-9823 arm: imx: correct L2 controller settings
 after resume

As we have specific tag and data latency settings on our platforms,
so we have to restore these settings after resume with L2 controller
power gated. Otherwise, system perpormance will be impacted a lot:

dd read test(dd if=/dev/mmcblk2 of=/dev/null bs=1M count=2000) of SD
card would lower from 61.4MB/s to 57.7MB/s, ~6% drop.

Signed-off-by: Anson Huang <b20788@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 arch/arm/mach-imx/headsmp.S |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-imx/headsmp.S b/arch/arm/mach-imx/headsmp.S
index 5456837..2c61117 100644
--- a/arch/arm/mach-imx/headsmp.S
+++ b/arch/arm/mach-imx/headsmp.S
@@ -65,6 +65,10 @@ ENDPROC(v7_secondary_startup)
 	ldr	r0, [r2, #L2X0_R_PHY_BASE]	@ get physical base of l2x0
 	ldr	r1, [r2, #L2X0_R_AUX_CTRL]	@ get aux_ctrl value
 	str	r1, [r0, #L2X0_AUX_CTRL]	@ restore aux_ctrl
+	ldr	r1, [r2, #L2X0_R_TAG_LATENCY]		@ get tag latency value
+	str	r1, [r0, #L2X0_TAG_LATENCY_CTRL]	@ restore tag latency
+	ldr	r1, [r2, #L2X0_R_DATA_LATENCY]		@ get data latency value
+	str	r1, [r0, #L2X0_DATA_LATENCY_CTRL]	@ restore data latency
 	mov	r1, #0x1
 	str	r1, [r0, #L2X0_CTRL]		@ re-enable L2
 	.endm
-- 
1.7.5.4

